### PR TITLE
Flagging fixes

### PR DESCRIPTION
**Pull request by abidlabs**
This PR fixes a few things currently broken with flagging:

* Only show flagging button if `allow_flagging == "manual"` (Closes: #1086)
* Fixes flagging audio files if they are loaded from examples (Closes: #841)

### CODE REVIEW

**Comment by abidlabs** [comment]
Ready to be reviewed @FarukOzderim @aliabid94

File: gradio/interface.py [hunk 1]
```diff
@@ -546,7 +546,15 @@ def clean_html(raw_html):
                     for component in self.output_components:
                         component.render()
                     with Row():
-                        flag_btn = Button("Flag")
+                        if self.allow_flagging == "manual":
+                            flag_btn = Button("Flag")
+                            flag_btn._click_no_preprocess(
+                                lambda *flag_data: self.flagging_callback.flag(
+                                    flag_data
```
Comment (FarukOzderim) [inline gradio/interface.py:553]: Not sure about this, but wouldn't this be the correct version:

```suggestion
                                    *flag_data
```

File: gradio/routes.py [hunk 2]
```diff
@@ -56,30 +56,6 @@ def render(self, content: Any) -> bytes:
 ###########
 
 
-class PredictBody(BaseModel):
```
Comment (FarukOzderim) [inline gradio/routes.py:59]: Why did you remove these validations?

Better update them instead of removing.

**Review by FarukOzderim — COMMENTED** [review]
LGMT other than the above points

File: gradio/routes.py (hunk 2 above)
Comment (abidlabs) [reply gradio/routes.py:59]: Good point, will update

File: gradio/interface.py (hunk 1 above)
Comment (abidlabs) [reply gradio/interface.py:553]: Nope `flagging_callback.flag()` takes a list for its first parameter, which is what `flag_data` is. `*flag_data` would unravel the list into separate parameters, which is not what we want here.

**Comment by abidlabs** [comment]
Thanks @FarukOzderim for the review. Have added the response validation back
